src/components/visualizations/chart/chartOptionsBuilder.ts

```diff
@@ -381,23 +322,45 @@ export function getSeriesItemData(
     });
 }
 
-export interface ISeriesItemConfig {
-    color: string;
-    legendIndex: number;
-    data?: any;
-    name?: string;
-    yAxis?: number;
-    xAxis?: number;
-}
-
 export function getHeatmapSeries(
     executionResultData: Execution.DataValue[][],
     measureGroup: Execution.IMeasureGroupHeader["measureGroupHeader"],
 ) {
-    const data = [] as any;
-    executionResultData.forEach((rowItem: any, rowItemIndex: number) => {
-        rowItem.forEach((columnItem: any, columnItemIndex: number) => {
-            data.push({ x: columnItemIndex, y: rowItemIndex, value: parseValue(columnItem) });
+    const data: IPointData[] = [];
+    const nullColor: IPatternObject = {
+        pattern: {
+            path: {
+                d: "M 10 0 L 0 10 M 9 11 L 11 9 M 4 11 L 11 4 M -1 1 L 1 -1 M -1 6 L 6 -1",
+                stroke: GRAY,
+                strokeWidth: 1,
+                fill: WHITE,
+            },
+            width: 10,
+            height: 10,
+        },
+    };
+    let value: number;
+    let pointData: IPointData;
+    executionResultData.forEach((rowItem: Execution.DataValue[], rowItemIndex: number) => {
+        rowItem.forEach((columnItem: Execution.DataValue, columnItemIndex: number) => {
+            value = parseValue(columnItem + "");
```

Comment:
are you trying to force type conversion with `+ ""`? if yes please find a solution which will avoid usage of this javascript "magic" :)

Reply:
`Execution.DataValue` is combined from `number|string|null` but `parseValue` only accept `string`.
It can be rewrite as `value = typeof columnItem === "number" ? columnItem : parseValue(columnItem + "");`
But `value = parseValue(columnItem + "");` is shorter and clear enough.

Reply:
http://2ality.com/2012/03/converting-to-string.html please use `String(columnItem)` :)

Reply:
done